src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -2306,13 +2306,6 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
                 RaisePropertyChanged(nameof(ArgumentLacing));
                 RaisePropertyChanged(nameof(IsVisible));
                 RaisePropertyChanged(nameof(DisplayLabels));
-                RaisePropertyChanged(nameof(IsSetAsInput));
-                RaisePropertyChanged(nameof(IsSetAsOutput));
-                //we need to modify the downstream nodes manually in case the
-                //undo is for toggling freeze. This is ONLY modifying the execution hint.
-                // this does not run the graph.
-                RaisePropertyChanged(nameof(IsFrozen));
-                MarkDownStreamNodesAsModified(this);
```

Comment:
Hi @mmisol Did you test that downstream node can be also reverted back to the previous state?

Reply:
I'm not sure I follow. Could you mention the steps of the test?

Reply:
Sure, if a node is frozen, the downstream nodes are also frozen. The original code seems taking care of correctly marking downstream nodes when undo freezing node as well. Just want to make sure the behavior is the same.

Reply:
Oh ok. Yes, I tested that and it works